LexSample/app/src/main/java/com/amazonaws/sample/lex/TextActivity.java

```diff
@@ -143,33 +144,68 @@ public boolean onKey(View v, int keyCode, KeyEvent event) {
     private void initializeLexSDK() {
         Log.d(TAG, "Lex Client");
 
+        final CountDownLatch initializeLatch = new CountDownLatch(1);
+        final CountDownLatch getCredentialsLatch = new CountDownLatch(1);
+
         // Initialize the mobile client
         AWSMobileClient.getInstance().initialize(this, new Callback<UserStateDetails>() {
             @Override
             public void onResult(UserStateDetails result) {
-                Log.d(TAG, "onResult: ");
-                // Create Lex interaction client.
-                lexInteractionClient = new InteractionClient(getApplicationContext(),
-                        AWSMobileClient.getInstance(),
-                        Regions.fromName(appContext.getResources().getString(R.string.lex_region)),
-                        appContext.getResources().getString(R.string.bot_name),
-                        appContext.getResources().getString(R.string.bot_alias));
-                lexInteractionClient.setAudioPlaybackListener(audioPlaybackListener);
-                lexInteractionClient.setInteractionListener(interactionListener);
+                Log.d(TAG, "initialize.onResult, userState: " + result.getUserState().toString());
+                initializeLatch.countDown();
             }
 
             @Override
             public void onError(Exception e) {
-                Log.e(TAG, "onError: ", e);
+                Log.e(TAG, "initialize.onError: ", e);
+                initializeLatch.countDown();
             }
         });
+
+        try {
+            initializeLatch.await();
+
+            // Identity ID is not available until we make a call to get credentials, which also
+            // caches identity ID.
+            AWSMobileClient.getInstance().getAWSCredentials(new Callback<AWSCredentials>() {
```

Comment:
In case of AWSMobileClient initialization error, we will be counting down the latch and getAWSCredentials call will result in NPE. We might consider adding a check for success initialization of AWSMobileClient